### ISSUE TITLE
Incorrect Implementation of describeContents() in LoyaltyCard Class

### ISSUE THREAD

**Issue by Aglag257** [CatimaLoyalty/Android#1849]
The `describeContents()` method in the `LoyaltyCard` class currently returns the `id` of the card. According to the Android documentation, `describeContents()` should return a bitmask indicating the set of special object types marshaled by the `Parcelable` instance. The typical return value is `0`, as most implementations do not have any special characteristics.

### Expected Behavior

`describeContents()` should return `0` unless the `Parcelable` includes a file descriptor or other special objects, which would be indicated by the bitmask constant `CONTENTS_FILE_DESCRIPTOR`.

### Actual Behavior

The method returns the `id` of the `LoyaltyCard`, which does not align with the expected behavior as per Android's `Parcelable` interface contract.

### Steps to Reproduce

Review the implementation of `describeContents()` in `LoyaltyCard.java`:

```java
@Override
public int describeContents() {
    return id;
}
```

### Suggested Fix

Update the `describeContents()` method in `LoyaltyCard.java` to return `0`:

```java
@Override
public int describeContents() {
    return 0;
}
```

### Supporting Documentation

Here is the relevant section of the Android developer documentation that describe the `describeContents()` method and its intended use:

- [Parcelable interface documentation](https://developer.android.com/reference/android/os/Parcelable#describeContents())

Please update the method to conform with the standard `Parcelable` implementation practices or let me know if i misunderstood something

Thank you for considering this issue.

**Comment by TheLastProject** [comment]
Sounds like you're probably right. Looks like this was added 3 years ago to fix a crash on rotation: https://github.com/CatimaLoyalty/Android/commit/5ddcb0a66841e19d59b68f17df59a4eb1dd432bf

If you can test and confirm that your change won't break that it makes sense to change that so feel free to make a MR :)

**Comment by Aglag257** [comment]
Thank you for the quick response and for providing the commit it was added.

I will test the change and follow through with the MR

**Comment by obfusk** [comment]
Fixed by  #1887, closing.
